test/e2e/storage/testsuites/base.go

```diff
@@ -66,12 +66,12 @@ func getTestNameStr(suite TestSuite, pattern testpatterns.TestPattern) string {
 }
 
 // RunTestSuite runs all testpatterns of all testSuites for a driver
-func RunTestSuite(f *framework.Framework, config framework.VolumeTestConfig, driver drivers.TestDriver, tsInits []func() TestSuite) {
+func RunTestSuite(f *framework.Framework, config framework.VolumeTestConfig, driver drivers.TestDriver, tsInits []func() TestSuite, tunePatternFunc func([]testpatterns.TestPattern) []testpatterns.TestPattern) {
```

Comment:
Is framework.VolumeTestConfig a good choice here? It has lots of options that don't seem to be used anymore, or at least don't have an effect. For example, provisioning.go only uses ClientNodeName but ignores NodeSelector.

Do you think it might be clear to have a testsuite.Config which has just the options that really matter for the testsuite package?

Reply:
Good points. 

This framework has created in a way that existing tests can be called in a common way for any drivers. (These tests were originally tightly bound to particular driver's particular source, like volume source, pv source, or storage class. And this PR was aimed to separate them to drivers and testsuites, and make testsuites called in a common loop.) 

In the process, I tried to keep the existing tests as much as they were so that I won't break how tests were working. In other word, how the test is called was changed, but how each test works haven't been improved, yet. 

Now, above aim was achieved, so we are ready to consider improving how each test works. Defining a new testsuite.Config that can be commonly used in all testsuites will be one of the remaining items on how each test works. And it will be helpful in adding more testsuites easily without bugs in the future.